scoper.inc.php

```diff
@@ -102,18 +102,9 @@ function ($filePath, $prefix, $contents) {
 
             return $contents;
         },
-        function ($filePath, $prefix, $contents) {
-            $ret = str_replace(
-                $prefix . '\Psalm\Plugin\\',
-                'Psalm\Plugin\\',
-                $contents
-            );
-            return $ret;
-        },
     ],
     'whitelist' => [
         \Composer\Autoload\ClassLoader::class,
-        'Psalm\Plugin\*',
```

Comment:
Wouldn't these changes cause problems running plugins with phar-packaged Psalm?

Reply:
Probably - I was just seeing errors when running on Psalm’s own codebase, and assumed the errors were universal. I’ll remove tomorrow

Reply:
Do plugins work with the Phar at all?

Reply:
Partially. For example, PHPUnit plugin loads, but will probably fail on some versions where it accesses internal classes. Plugins can work with phar, given enough care both from the plugin and Psalm's sides. As long as phar build does not mangle any class names plugin may use it should be working just fine.

Reply:
Yeah, but the plugin hooks demand non-namespaced things like `Psalm\StatementsSource`

Reply:
- Not every plugin may have hooks (doctrine plugin is still stubs-only, but that's going to change)
- My original idea was to provide interfaces in `Plugin` namepsace for everything plugins may need to depend on